source/Halibut.Tests/Support/Streams/DelegateStreamBase.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Halibut.Tests.Support.Streams
+{
+    public abstract class DelegateStreamBase :  Stream
+    {
+        public abstract Stream Inner { get; }
```

Comment:
Nit: Could make this an actual property (protected even?), so we don't have to implement it